zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala

```diff
@@ -1866,4 +1871,592 @@ object DynamicValue {
+        transform => transform.value
+      ),
+      _.asInstanceOf[DynamicValue.Transform]
+    )
+
+  private val someValueCase: Schema.Case[DynamicValue.SomeValue, DynamicValue] =
+    Schema.Case(
+      "SomeValue",
+      Schema.CaseClass1[DynamicValue, DynamicValue.SomeValue](
+        Schema.Field("value", Schema.defer(DynamicValueSchema())),
+        dv => DynamicValue.SomeValue(dv),
+        someValue => someValue.value
+      ),
+      _.asInstanceOf[DynamicValue.SomeValue]
+    )
+
+  private val dictionaryCase: Schema.Case[DynamicValue.Dictionary, DynamicValue] =
+    Schema.Case(
+      "Dictionary",
+      Schema.CaseClass1[Chunk[(DynamicValue, DynamicValue)], DynamicValue.Dictionary](
+        Schema.Field(
+          "entries",
+          Schema.defer(Schema.chunk(Schema.tuple2(DynamicValueSchema(), DynamicValueSchema())))
+        ),
+        chunk => DynamicValue.Dictionary(chunk),
+        dictionary => dictionary.entries
+      ),
+      _.asInstanceOf[DynamicValue.Dictionary]
+    )
+
+  private val sequenceCase: Schema.Case[DynamicValue.Sequence, DynamicValue] =
+    Schema.Case(
+      "Sequence",
+      Schema.CaseClass1[Chunk[DynamicValue], DynamicValue.Sequence](
+        Schema.Field("values", Schema.defer(Schema.chunk(DynamicValueSchema()))),
+        chunk => DynamicValue.Sequence(chunk),
+        seq => seq.values
+      ),
+      _.asInstanceOf[DynamicValue.Sequence]
+    )
+
+  private val enumerationCase: Schema.Case[DynamicValue.Enumeration, DynamicValue] =
+    Schema.Case(
+      "Enumeration",
+      Schema.CaseClass1[(String, DynamicValue), DynamicValue.Enumeration](
+        Schema.Field(
+          "value",
+          Schema.defer(Schema.tuple2(Schema.primitive(StandardType.StringType), DynamicValueSchema()))
+        ),
+        value => DynamicValue.Enumeration(value),
+        enumeration => enumeration.value
+      ),
+      _.asInstanceOf[DynamicValue.Enumeration]
+    )
+
+  private val recordCase: Schema.Case[DynamicValue.Record, DynamicValue] =
+    Schema.Case(
+      "Record",
+      Schema.CaseClass1[Map[String, DynamicValue], DynamicValue.Record](
+        Schema.Field(
+          "values",
+          Schema.defer(Schema.map(Schema.primitive(StandardType.StringType), DynamicValueSchema()))
+        ),
+        map => DynamicValue.Record(map.asInstanceOf[ListMap[String, DynamicValue]]),
+        record => record.values
+      ),
+      _.asInstanceOf[DynamicValue.Record]
+    )
+
+  private val dynamicAstCase: Schema.Case[DynamicValue.DynamicAst, DynamicValue] =
+    Schema.Case(
+      "DynamicAst",
+      Schema.CaseClass1[SchemaAst, DynamicValue.DynamicAst](
+        Schema.Field("ast", SchemaAst.schema),
+        schemaAst => DynamicValue.DynamicAst(schemaAst),
+        dynamicAst => dynamicAst.ast
+      ),
+      _.asInstanceOf[DynamicValue.DynamicAst]
+    )
+
+//  private def singletonCase[A]: Schema.Case[DynamicValue.Singleton[A], DynamicValue] = ???
```

Comment:
For this one I think we just use a case for `Singleton[Unit]`. The singleton Schema isomorphic to `Unit` so when we turn a `DynamicValue` into an `A` using `Schema[A]` we get the singleton instance from the `Schema` itself, so there is nothing to encode in the `DynamicValue`. So I think this one can just be:

```scala
Schema.Case[DynamicValue.Singleton[Any], DynamicValue](
  "Unit",
  Schema[Unit].transform(_ => DynamicValue.Singleton(()), _ => ()),
 _.asInstanceOf[DynamicValue.Singleton[Any]]
)
```

Reply:
> Hey @thinkharderdev!
> I more or less figured out how to deal with `DynamicValue.Primitive[A]`.
> Basially `Primitive[A]` is limited by the number of `StandardType[A]` values, so I listed all Schema.Case for Primitive for which there are defined StandardType values.
> However, I have no idea what to do with `DynamicValue.Singleton[A]` though...
> @jdegoes Hey John! Maybe you have some thoughts on making schema for `DynamicValue.Singleton[A]`?

Awesome! Left a couple comments that I think should help address these points!